libraries/AP_NavEKF3/AP_NavEKF3_PosVelFusion.cpp

```diff
@@ -981,7 +993,9 @@ void NavEKF3_core::FuseVelPosNED()
         if (fuseVelData) {
             fuseData[0] = true;
             fuseData[1] = true;
-            if (useGpsVertVel || useExtNavVel) {
+            if (useGpsVertVel && frontend->sources.useVelZSource(AP_NavEKF_Source::SourceZ::GPS)) {
```

Comment:
I'm surprised this isn't just using fuseVelVertData

Reply:
fixed